templates/common/templatesource/download.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package common contains the common utility functions for template commands.
+
+package templatesource
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/pkg/logging"
+)
+
+const (
+	// These will be used as part of the names of the temporary directories to
+	// make them identifiable.
+	templateDirNamePart = "template-copy-"
+)
+
+type RunParams struct {
```

Comment:
This should be removed since the last refactor, right?

Reply:
removed. Sorry that I missed this.